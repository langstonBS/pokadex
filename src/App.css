.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

ul {
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.search {
  display: flex;
  justify-content: center;
  padding: 10px;
}

button{
  background-color: blueviolet;
  border: none;
  color: white;
  padding: 5px 30px;
  text-align: center;
  display: inline-block;
  font-size: 16px;
  margin: 15px;
  border-radius: 5px;
  transition-duration: 0.4s;
}
button:hover{
  background-color: rgb(70, 13, 122);
}
.searchList {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  background-color: blueviolet;
  padding-bottom: 20px;
  border-radius: 10px 10px 0 0;
  
}

select {
  position: relative;
  display: flex;
  width: 10em;
  height: 2em;
  line-height: 2;
  border-radius: .25em;
  transition-duration: 0.4s;
}

