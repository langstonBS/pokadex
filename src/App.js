import React, { Component } from 'react';
//import request from 'superagent';
import RenderPokemon from './RenderPokemon';

export default class App extends Component {

  state = {
    searchQuery: null,
    searchLevle: 0,
    getData: null,
    attacDefenc: 'attack',
  }
  handleChange = (e) => {
    const value = e.target.value;
    this.setState({ searchQuery: value });
    console.log(this.state)
  }
  levelChange = (e) => {
    let value = e.target.value;
    if (value <= 0|| value - (Math.floor(value)) !== 0) {
      value = 0;
    }
    this.setState({ searchLevle: value });
    console.log(this.state)
  }
  sortBy = (e) => {
    const value = e.target.value;
    this.setState({ attacDefenc: value })
    console.log(value);

  }

  handleClick = () => {
    const getData = 'I am clicked';
    console.log(getData);
}

  render() {
    return (
      <div>
        i am here
        <input onChange={this.handleChange} />
        <input type="number" pattern="[0-9]" onChange={this.levelChange} />
        <select className="selctionStyle" onChange={this.sortBy}>
          <option value="attack">attack</option>
          <option value="defend">defend</option>
        </select>
        < button onClick={this.handleClick}>serch</button>
        < RenderPokemon />
      </div>
    )
  }
}
