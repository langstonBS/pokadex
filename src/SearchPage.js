import React, { Component } from 'react';
import request from 'superagent';
import RenderPokemon from './RenderPokemon';
import SerchQurryPage from './SerchQurryPage';
import './App.css';


export default class SearchPage extends Component {
  state = {
    searchQuery: '',
    searchLevle: 0,
    data: [{}],
    attacDefenc: 'attack',
    ascDesc: 'asc',
    page: 1,
    typePram: '',
    dataType: [],
    pageInfo: 0
  }

  async componentDidMount() {
    const searchParams = new URLSearchParams(window.location.search);
    const pageQuery = searchParams.get('page');
    const pokemonQuery = searchParams.get('pokemon')
    console.log(pokemonQuery);
    if (pokemonQuery){
      this.setState({ searchQuery: pokemonQuery })
    }
    if (pageQuery) {
      let startPage = 1;
      if (searchParams.get('page')) {
        startPage = searchParams.get('page');
      }
      this.setState({ page:  pageQuery})
      await this.serchMethod(startPage)
    
    } else {
      const page = this.state.page;
      await this.serchMethod(page);
    }
    await this.serchMethod();
    const fetchedDatabar = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex/types`);
    this.setState({ dataType: fetchedDatabar.body });
}

  handleChange = (e) => {
    const value = e.target.value;
    this.setState({ searchQuery: value });
  };

  levelChange = (e) => {
    let value = e.target.value;
    if (value <= 0 || value - (Math.floor(value)) !== 0) {
      value = 0;
    }
    this.setState({ searchLevle: value });
  };

  sortBy = (e) => {
    const value = e.target.value;
    this.setState({ attacDefenc: value })
  }

  asendingDesending = (e) => {
    const value = e.target.value;
    this.setState({ ascDesc: value })
  }

  handleClick = async () => {
    await this.serchMethod();
  }

  pagePlacePrevus = async () => {
    let curentPage = this.state.page;
    let Newpage
    if (Number(curentPage) === 1) {
      Newpage = Number(curentPage);
    } else {
      Newpage = Number(curentPage) - 1;
    }
    this.setState({ page: Newpage });
    await this.serchMethod(Newpage);
  }

  tyepofPokemon = (e) => {
    const value = e.target.value;
    this.setState({ typePram: value });
  };
  
  pagePlaceNext = async () => {
    console.log("before page", this.state.pageInfo);
    let curentPage = this.state.page;
    console.log("cuernt page befor", curentPage);
    let Newpage = 0;
    let pokecount = this.state.pageInfo.count;
    let perPageCount = this.state.pageInfo.perPage;
    let pageLeft = pokecount - (curentPage * perPageCount);
    console.log("cuernt page befor", curentPage);  
    if (pageLeft >= 1) {
      Newpage = Number(curentPage) + 1;
    } else {
      Newpage = Number(curentPage);
    }
    console.log("cuernt page after", curentPage);
    console.log("newpage page after", Newpage);
    this.setState({ page: Newpage });
    await this.serchMethod(Newpage);
}

  async serchMethod(page) {
    if (!page) {
      page = this.state.page; 
    }
    console.log('page', page);
  let fetchedData = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?pokemon=${this.state.searchQuery}&perPage=20&type_1=${this.state.typePram}&sort=${this.state.attacDefenc}&direction=${this.state.ascDesc}&page=${page}`);
    this.setState({ data: fetchedData.body.results });
    this.setState({ pageInfo: fetchedData.body });
    console.log(this.state.pageInfo);
    
}

  render() {
    return (
      <>
       
        <SerchQurryPage
          handleChange={this.handleChange}
          levelChange={this.levelChange}
          sortBy={this.sortBy}
          handleClick={this.handleClick}
          asendingDesending={this.asendingDesending}
          tyepofPokemon={this.tyepofPokemon}
          pagePlacePrevus={this.pagePlacePrevus}
          pagePlaceNext={this.pagePlaceNext}
        />
      <span>  
        <ul>
          {this.state.data.map(item => <RenderPokemon pokemon={item} />)}
        </ul>
        </span>
      </>
    )
  }
}
