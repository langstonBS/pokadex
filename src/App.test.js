// import React from 'react';
// import { render } from '@testing-library/react';
import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchPage from './SearchPage'
import DetailPage from  './PokemonDetail'

describe('First React component test with Enzyme', () => {
   it('renders without crashing', () => {
     const wrapper = shallow(<App />);
     expect(wrapper).toMatchSnapshot()
    });
});

describe('First React component test with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot()
   });
});


  it('renders without crashing', () => {
    const wrapper = shallow(<DetailPage />);
    expect(wrapper).toMatchSnapshot()
   });

