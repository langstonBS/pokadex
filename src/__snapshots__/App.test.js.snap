// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;

exports[`First React component test with Enzyme renders without crashing 2`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
